src/include/serial.i

```diff
@@ -309,11 +310,20 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_PAGE *page,
 	 * is used as an indicator of there being further updates on this page.
 	 */
 	if ((txn = page->modify->obsolete_check_txn) != WT_TXN_NONE) {
-		if (!__wt_txn_visible_all(session, txn, NULL)) {
+#ifdef HAVE_TIMESTAMPS
+		obsolete_timestamp =
+		    page->modify->has_obsolete_check_timestamp ?
+		    &page->modify->obsolete_check_timestamp : NULL;
+#else
+		WT_UNUSED(obsolete_timestamp);
+#endif
```

Comment:
I think this block can be simplified to `obsolete_timestamp = WT_TIMESTAMP_NULL(&page->modify->obsolete_check_timestamp);`, then pass it unconditionally below.